x-pack/plugins/transform/server/usage/collector.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UsageCollectionSetup } from '@kbn/usage-collection-plugin/server';
+import { TransformHealthRuleParams } from '../../common/types/alerting';
+import { TRANSFORM_RULE_TYPE } from '../../common';
+
+export interface TransformAlertsUsageData {
+  alertRules: {
+    transform_health: {
+      count_by_check_type: {
+        notStarted: number;
+        errorMessages: number;
+        healthCheck: number;
+      };
+    };
+  };
+}
+
+export function registerCollector(usageCollection: UsageCollectionSetup, alertIndex: string) {
+  const collector = usageCollection.makeUsageCollector<TransformAlertsUsageData>({
+    type: 'transform',
+    schema: {
+      alertRules: {
+        transform_health: {
+          count_by_check_type: {
+            notStarted: {
+              type: 'long',
+              _meta: {
+                description:
+                  'total number of alerting rules performing the not started health check',
+              },
+            },
+            errorMessages: {
+              type: 'long',
+              _meta: {
+                description:
+                  'total number of alerting rules performing the error message health check',
+              },
+            },
+            healthCheck: {
+              type: 'long',
+              _meta: {
+                description:
+                  'total number of alerting rules performing the health check with the stats API',
+              },
+            },
+          },
+        },
+      },
+    },
+    isReady: () => true,
+    fetch: async ({ esClient }) => {
+      const transformHealthRuleInstances = await esClient.search<{
+        alert: {
+          params: TransformHealthRuleParams;
+        };
+      }>(
+        {
+          index: alertIndex,
+          size: 10000,
+          query: {
+            bool: {
+              filter: [
+                { term: { type: 'alert' } },
+                {
+                  term: {
+                    'alert.alertTypeId': TRANSFORM_RULE_TYPE.TRANSFORM_HEALTH,
+                  },
+                },
+              ],
+            },
+          },
+        },
+        { maxRetries: 0 }
+      );
+
+      const resultsByCheckType = transformHealthRuleInstances.hits.hits.reduce(
+        (acc, curr) => {
+          const doc = curr._source;
+          if (!doc) return acc;
+
+          const {
+            alert: {
+              params: { testsConfig },
```

Comment:
Testing this locally, I am seeing `testsConfig` being initialized as `null` when the rule is created and the switches are left to the default 'on' position, so counts show as `0`. 

```
            "transform": {
              "alertRules": {
                "transform_health": {
                  "count_by_check_type": {
                    "notStarted": 0,
                    "errorMessages": 0,
                    "healthCheck": 0
                  }
                }
              }
            },
```

It is only after switching the controls to 'off', and then back to 'on' and saving the edit that `testsConfig` is populated with
```
testsConfig: {
  healthCheck: { enabled: true },
  notStarted: { enabled: true },
  errorMessages: null
}
```

Reply:
Note that with the switches in the default 'on' positions, the rule is working correctly, it's just the counts in the telemetry config are all `0`.

Reply:
Fixed in https://github.com/elastic/kibana/pull/159652/commits/c0896071d313c543ff21ef545f3f2cef4f27d29a